BinarySearch/MedianOfArray.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+There are two sorted arrays A and B of size m and n respectively.
+
+Find the median of the two sorted arrays ( The median of the array formed by merging both the arrays ).
+
+The overall run time complexity should be O(log (m+n)).
+
+Sample Input
+
+A : [1 4 5]
+B : [2 3]
+
+Sample Output
+
+3
+
+ NOTE: IF the number of elements in the merged array is even, then the median is the average of n / 2 th and n/2 + 1th element.
+For example, if the array is [1 2 3 4], the median is (2 + 3) / 2.0 = 2.5
+
+https://www.interviewbit.com/problems/median-of-array/
+
+*/
+
+double Solution::findMedianSortedArrays(const vector<int> &A, const vector<int> &B) {
+    // Do not write main() function.
+    // Do not read input, instead use the arguments to the function.
+    // Do not print the output, instead return values as specified
+    // Still have a doubt. Checkout www.interviewbit.com/pages/sample_codes/ for more details
+    if(A.empty())
+        return (B.size()%2==0)? (B[B.size()/2]+B[B.size()/2-1])/2.0: B[B.size()/2];
+    if(B.empty())
+        return (A.size()%2==0)? (A[A.size()/2]+A[A.size()/2-1])/2.0: A[A.size()/2];
+    if(A.size()>B.size())
+        return findMedianSortedArrays(B,A);
+    int n = A.size(), m= B.size();
+    int start = 0,end = n,partitionX,partitionY,maxLeftX,maxLeftY,minRightX,minRightY;
+    while(start<=end)
+    {
+        partitionX = (start+end)/2;
+        partitionY = (n+m+1)/2 - partitionX;
+        
+        maxLeftX = (partitionX==0)?INT_MIN:A[partitionX-1];
```

Comment:
Please use `spaces` to separate logical components everywhere in this file. For example:
maxLeftX = (partitionX` `==` `0)` `?` `INT_MIN` `:` `A[partitionX` `-` `1];

This will automatically make the code more readable and easily understandable.